lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -492,6 +494,16 @@ def json_error(raw_response)
         }
       end
 
+      def response_is_test?(response)
+        if response.has_key?('livemode')
+          !response['livemode']
+        elsif response['charge'].is_a?(Hash) && response['charge'].has_key?('livemode')
+          !response['charge']['livemode']
+        else
+          false
```

Comment:
Logging would be nice, but there's no existing logging currently. Gonna keep this as is to avoid changing behaviour.